Cubical/HITs/PropositionalTruncation/Properties.agda

```diff
@@ -27,26 +29,41 @@ rec : ∀ {P : Type ℓ} → isProp P → (A → P) → ∥ A ∥ → P
 rec Pprop f ∣ x ∣ = f x
 rec Pprop f (squash x y i) = Pprop (rec Pprop f x) (rec Pprop f y) i
 
-propTruncIsProp : isProp ∥ A ∥
-propTruncIsProp x y = squash x y
-
 elim : ∀ {P : ∥ A ∥ → Type ℓ} → ((a : ∥ A ∥) → isProp (P a))
-  → ((x : A) → P ∣ x ∣) → (a : ∥ A ∥) → P a
+     → ((x : A) → P ∣ x ∣) → (a : ∥ A ∥) → P a
 elim Pprop f ∣ x ∣ = f x
-elim {A = A} {P = P} Pprop f (squash x y i) =
-  isOfHLevel→isOfHLevelDep 1 Pprop (elim Pprop f x) (elim Pprop f y) (squash x y) i
+elim Pprop f (squash x y i) =
+  isOfHLevel→isOfHLevelDep 1 Pprop
+    (elim Pprop f x) (elim Pprop f y) (squash x y) i
+
+elim2 : {P : ∥ A ∥ → ∥ A ∥ → Type ℓ}
+        (Bset : ((x y : ∥ A ∥) → isProp (P x y)))
+        (g : (a b : A) → P ∣ a ∣ ∣ b ∣)
+        (x y : ∥ A ∥) → P x y
+elim2 Pprop g = elim (λ _ → isPropΠ (λ _ → Pprop _ _))
+                     (λ a → elim (λ _ → Pprop _ _) (g a))
+
+elim3 : {P : ∥ A ∥ → ∥ A ∥ → ∥ A ∥ → Type ℓ}
+        (Bset : ((x y z : ∥ A ∥) → isProp (P x y z)))
+        (g : (a b c : A) → P (∣ a ∣) ∣ b ∣ ∣ c ∣)
+        (x y z : ∥ A ∥) → P x y z
+elim3 Pprop g = elim2 (λ _ _ → isPropΠ (λ _ → Pprop _ _ _))
+                      (λ a b → elim (λ _ → Pprop _ _ _) (g a b))
+
+propTruncIsProp : isProp ∥ A ∥
+propTruncIsProp x y = squash x y
 
-propId : isProp A → ∥ A ∥ ≡ A
-propId {A = A} hA = isoToPath (iso (rec hA (idfun A)) (λ x → ∣ x ∣) (λ _ → refl) rinv)
+propTruncId : isProp A → ∥ A ∥ ≡ A
+propTruncId {A = A} hA = isoToPath (iso (rec hA (idfun A)) (λ x → ∣ x ∣) (λ _ → refl) rinv)
```

Comment:
Should we export the iso too? also probably better to define it by copatterns.

Reply:
I'll export it as an equiv, but define the iso using copatterns in a where